docs/framework/30scoring/implementation.md

```diff
@@ -1,9 +1,51 @@
+The following aspects of [Implementation Risk](../20categories/30implementation/protocol-implementation-risk.md#protocol-implementation-risk) are currently not taken into account when assessing the Implementation Risk Score: 
+[Mixing of Control and Data Plane](../20categories/30implementation/mixing-control-data-flow.md), 
+[Role Based Access Control](../20categories/30implementation/access-control.md),
+[Upgradability](../20categories/30implementation/upgrade.md),
+[Secret Storage](../20categories/30implementation/secret-storage.md),
+[Product Development Maturity](../20categories/30implementation/maturity.md),
+[Well Known Platform](../20categories/30implementation/known-platform.md),
+[Well Known Smart Contract Programming Language](../20categories/30implementation/known-language.md),
+[Formal Verification](../20categories/30implementation/formal-verification.md),
+[Testing](../20categories/30implementation/testing.md),
+[Open Source Code](../20categories/30implementation/open-source.md),
+[Verified Code on Block Explorer](../20categories/30implementation/verified-code.md),
+[Documentation](../20categories/30implementation/documentation.md),
+[Bug Bounty](../20categories/30implementation/bug-bounty.md),
+[Ability to Pause Project](../20categories/30implementation/pause.md),
+and 
+[Ability to Ban Addresses](../20categories/30implementation/ban-address.md).
+
 
 The equation for the *Implementation Risk Score* is:
 
 ```
-Implementation Risk Score = 10
+Implementation Risk Score = Code Auditing Score
+```
+
+### Code Auditing Score
+Code auditing is described in detail in the [Code Auditing](../20categories/30implementation/audit.md) section. The equation for the *Code Auditing Score* is:
+
+```
+Code Audit Score = (I011 + I012 + I014 + I1015) x I013
 ```
+
+The *Code Auditing Score* ranges from 0 to 10.
+
+
+|Question ID  | Question                     |
+|-------------|------------------------------|
+| I011        | Has any version of the on-chain components of the system been audited? If yes, score 2. If no, score 0. |
+| I012        | Has any versions of the on-chain components of the system been audited by more than one audit organisation? If yes, score 2. If no, score 0. |
+| I013        | For the most recent audit, were all critical issues addressed? If yes, score 1. If no, score 0. |
+| I014        | Has the deployed version of the on-chain components of the system been audited? If yes, score 4. If no, score 0. |
+| I015        | Has any version of the off-chain components of the system been audited? If yes, score 2. If no, score 0. If the system contains no off-chain components, then score 2. |
+
+Rationale for scoring:
+
+* Unaddressed critical issues could indicate vulnerabilites.
+* Having an audit, addressing issues, and then deploying the revised code without re-auditing can result in vulnerabilities.
+* The quality of audit firms is not factored in the score as this is very subjective. 
```

Comment:
I agree that determining the quality of an audit firm can be somewhat subjective. However, I fear that equating the quality of audits from established players (e.g. Certik 3.5K projects with a combined current market cap of $364B) with those from new companies with little to no track record might conceal an important risk signal from our scoring. While we don't have to do it in this PR, I would suggest we think about a way to at least distinguish established from novice players based on their public track record.

Reply:
Could we have a "quality" factor: 
Audit firm has done fewer than 10 audits: Score 1.
Audit firm has done between 10 and fewer than 100 audits: Score 2.
Audit firm has done 100 or more audits: Score 3.

Or something like that?

Reply:
@ArjunChand13 and @ermyas  does that suggestion ^^^ sound good to you? If so, i will modify the PR to have text to that affect.

Reply:
This looks ok to me for now. A few thoughts below: 

I think that the credibility of an audit largely depends on the calibre of the auditor. Hence, we should perhaps weigh this factor a bit more. Having only a two-point difference between entities that have done thousands of audits vs those that are novices might not capture the importance of this metric well. 

It might be useful to capture whether the auditor has experience auditing cross-chain protocols, which might suggest better familiarity with crosschain architecture, common security pitfalls etc.

It might be useful to capture whether the scope of the audit includes execution environments, runtimes or language that the auditor does not have experience in (e.g. fewer auditors might have experience in non-EVM and non-Solidity environments)

Of course, I don't think we should try and capture every consideration in these risk scores, only the most salient ones, and where to draw the line in this regard can be tricky. 

I am not opposed to starting with the current proposal and iterating at a later point.

On the point assignment specifically: If there are multiple auditors, do we average out the _quality_ scores for each auditor or just consider the score of the most experienced?